### PR TITLE
configure global jwt bearer auth for swagger documentation

### LINKED ISSUE #4273: springdoc-jwt : authorization header didn't exist in requests
Hi,
I tried to use **springdoc-jwt** module on the dummy appplication

Then after my application started I pass the bearer token (got with the [/api/authenticate] endpoint with admin/admin credentials) into an authentication form with swagger UI:
image

When I invoke the getBeers enpoint :

**Actual result**:

curl -X 'GET' 'http://localhost:8080/api/beers' -H 'accept: */*'

Bearer token didn't exist in the request and the request fails with {"title": "Unauthorized",  "status": 401, "detail": "Full authentication is required to access this resource"}

**Expected result**:

curl -X 'GET' \
  'http://localhost:8080/api/beers' \
  -H 'accept: */*' \
  -H 'Authorization: Bearer eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJhZG1pbiIsImF1dGgiOiJST0xFX0FETUlOIiwiZXhwIjoxNjk5Mzk2Njg2fQ.qY_SdBezB1DL70w1gKLtu6UFgXwmtr5RFrfpfQkaAAdT7YXQkGwpPOTwnp8GghHGk7vjf9XaWYjz4VhG3x4iyw'

I have tried to fix that with adding  , security = @SecurityRequirement(name = "bearer-jwt") on each endpoint and It's working but I m not sure it's the good way

### PR DESCRIPTION
**Pull request by atomfrede**
This configures the swagger/openapi docs to use jwt bearer auth globally/for all endpoints

closes #4273

### CODE REVIEW

File: src/main/resources/generator/server/springboot/apidocumentation/springdocjwt/SpringdocJWTConfiguration.java.mustache [hunk 1]
```diff
@@ -27,7 +32,7 @@ class SpringdocJWTConfiguration {
         .scheme("Bearer")
         .bearerFormat("JWT")
         .in(SecurityScheme.In.HEADER)
-        .name("bearerAuth")
+        .name(securitySchemeName")
```
Comment (atomfrede) [inline src/main/resources/generator/server/springboot/apidocumentation/springdocjwt/SpringdocJWTConfiguration.java.mustache:35]:
```suggestion
        .name(securitySchemeName)
```

File: src/main/resources/generator/server/springboot/apidocumentation/springdocjwt/SpringdocJWTConfiguration.java.mustache [hunk 2]
```diff
@@ -14,9 +14,14 @@ import {{packageName}}.common.domain.Generated;
...
 
+  private final String securitySchemeName = "bearerAuth";
+
   @Bean
   GlobalOpenApiCustomizer jwtOpenApi() {
-    return openApi -> openApi.components(jwtComponents(openApi.getComponents()));
+    return openApi -> openApi
+        .addSecurityItem(new SecurityRequirement()
+            .addList(securitySchemeName))
+        .components(jwtComponents(openApi.getComponents()));
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/apidocumentation/springdocjwt/SpringdocJWTConfiguration.java.mustache:24]:
```suggestion
    return openApi ->
      openApi.addSecurityItem(new SecurityRequirement().addList(securitySchemeName)).components(jwtComponents(openApi.getComponents()));
```

**Review by pascalgrimaud — DISMISSED** [review]

File: src/main/resources/generator/server/springboot/apidocumentation/springdocjwt/SpringdocJWTConfiguration.java.mustache [hunk 3]
```diff
@@ -14,9 +15,12 @@ import {{packageName}}.common.domain.Generated;
 @Generated(reason = "Not called by integration tests")
 class SpringdocJWTConfiguration {
 
+  private final String securitySchemeName = "bearerAuth";
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/apidocumentation/springdocjwt/SpringdocJWTConfiguration.java.mustache:18]: Should be static (this is why sonar analysis is failling)

**Comment by DamnClin** [comment]
Tried it and still don't have the token sent to WebServices with 

```
{
  "actions" : [ {
    "module" : "init",
    "date" : "2022-11-12T13:55:21.420308450Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  }, {
    "module" : "maven-java",
    "date" : "2022-11-12T13:55:21.450095585Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  }, {
    "module" : "java-base",
    "date" : "2022-11-12T13:55:21.461004717Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  }, {
    "module" : "spring-boot",
    "date" : "2022-11-12T13:55:21.502165751Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  }, {
    "module" : "spring-boot-tomcat",
    "date" : "2022-11-12T13:55:21.524613582Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  }, {
    "module" : "spring-boot-cucumber",
    "date" : "2022-11-12T13:55:21.559082571Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  }, {
    "module" : "springdoc-mvc-openapi",
    "date" : "2022-11-12T13:55:21.583782452Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  }, {
    "module" : "zalando-problems",
    "date" : "2022-11-12T13:55:21.603707456Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  }, {
    "module" : "spring-boot-jwt",
    "date" : "2022-11-12T13:55:21.638204162Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  }, {
    "module" : "springdoc-jwt",
    "date" : "2022-11-12T13:55:21.659079173Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  }, {
    "module" : "spring-boot-jwt-basic-auth",
    "date" : "2022-11-12T13:55:21.676035909Z",
    "properties" : {
      "packageName" : "com.mycompany.myapp",
      "projectName" : "JHipster Sample Application",
      "baseName" : "jhipsterSampleApplication",
      "serverPort" : 8080,
      "endOfLine" : "lf",
      "indentSize" : 2
    }
  } ]
}
```

**Comment by atomfrede** [comment]
Let me check. It is supposed to do the same as adding the security requirement annotation to each endpoint.

**Comment by atomfrede** [comment]
I'll try to pick up this during this week

**Comment by DamnClin** [comment]
This behavior change with https://github.com/jhipster/jhipster-lite/pull/4586 (but no idea if it's better or worse...). When you query a WebService that need authentication you'll have a browser prompt with authentication, you can authenticate here and it'll work (this is the behavior for any autenticated endpoint). Is it ok ? IF so we need to remove the "Authorize" button wich become pretty useless

**Comment by pascalgrimaud** [comment]
closing in favor of https://github.com/jhipster/jhipster-lite/pull/5748

**Pull request closed without merging**